pycparser/c_parser.py

```diff
@@ -118,12 +139,16 @@ def __init__(
         # saw: int name;
         # If 'name' is not a key in _scope_stack[n] then 'name' was not defined
         # in this scope at all.
-        self._scope_stack = [dict()]
+
+        # typecheck input and deepcopy twice so we can reset to original
+        # identifiers in parse()
+        self._identifiers = dict([(str(key), bool(val)) for key, val in identifiers.items()])
+        self._scope_stack = [dict(self._identifiers.items())]
 
         # Keeps track of the last token given to yacc (the lookahead token)
         self._last_yielded_token = None
 
-    def parse(self, text, filename='', debuglevel=0):
+    def parse(self, text, filename='', debuglevel=0, reset_identifiers = True):
```

Comment:
Really depends on the use case... if it still were "parameter_context" specific as before, I'd agree, but for a generic interface I think it's useful to have the parameter. With reset_identifiers=False, it's possible to chain together multiple loads without using `get_context_parser()`. But when actually using `get_context_parser()` the reset_identifiers=True behaviour is much more likely to be useful.